src/document.rs

```diff
@@ -15,9 +15,37 @@ use tantivy as tv;
 
 use crate::{facet::Facet, to_pyerr};
 use pyo3::{PyMappingProtocol, PyObjectProtocol};
-use std::{collections::BTreeMap, fmt};
+use serde_json::Value as JsonValue;
+use std::{
+    collections::{BTreeMap, HashMap},
+    fmt,
+};
 use tantivy::schema::Value;
 
+fn value_to_object(val: &JsonValue, py: Python<'_>) -> PyObject {
+    match val {
+        JsonValue::Null => py.None(),
+        JsonValue::Bool(b) => b.to_object(py),
+        JsonValue::Number(n) => {
+            let oi64 = n.as_i64().map(|i| i.to_object(py));
+            let ou64 = n.as_u64().map(|i| i.to_object(py));
+            let of64 = n.as_f64().map(|i| i.to_object(py));
+            oi64.or(ou64).or(of64).expect("number too large")
```

Comment:
I feel like there is probably a cleaner way to do this, unfortunately, im unable to test this right now.

Reply:
I just copied code from this answer: https://stackoverflow.com/questions/70193869/exporting-hashmap-of-hashmap-to-python

Reply:
I think we probably want to try make this lazy if we can, bexause right now we're potentially calling python 3 or 4 times or so for a single value, which is quite high!

Reply:
How about now? Checked with is_i64, is_u64 & is_f64 and then turn to object py. In fact, I'm not sure if it's necessary to use `map(|i| i.to_object(py))` like before.